master/static/migrations/20231116155837_add-metrics-trigger.tx.up.sql

```diff
@@ -69,135 +63,28 @@ BEGIN
-                        FOR EACH ROW EXECUTE PROCEDURE stream_metric_change();
-
-DROP TRIGGER IF EXISTS stream_metric_raw_validations_trigger_iu ON raw_validations;
-CREATE TRIGGER stream_metric_raw_validations_trigger_iu
-    AFTER INSERT OR UPDATE
-                        ON raw_validations
-                        FOR EACH ROW EXECUTE PROCEDURE stream_metric_change();
-
-DROP TRIGGER IF EXISTS stream_metric_generic_metrics_trigger_iu ON generic_metrics;
-CREATE TRIGGER stream_metric_generic_metrics_trigger_iu
-    AFTER INSERT OR UPDATE
-                        ON generic_metrics
-                        FOR EACH ROW EXECUTE PROCEDURE stream_metric_change();
-
--- TODO(corban): replaces these triggers with one on the metrics table if we move to postgres >=11
--- DELETE should fire BEFORE to guarantee the experiment still exists to grab the workspace_id.
-DROP TRIGGER IF EXISTS stream_metric_raw_steps_trigger_d ON raw_steps;
-CREATE TRIGGER stream_metric_raw_steps_trigger_d
-    BEFORE DELETE ON raw_steps
-    FOR EACH ROW EXECUTE PROCEDURE stream_metric_change();
-
-DROP TRIGGER IF EXISTS stream_metric_raw_validations_trigger_d ON raw_validations;
-CREATE TRIGGER stream_metric_raw_validations_trigger_d
-    BEFORE DELETE ON raw_validations
-    FOR EACH ROW EXECUTE PROCEDURE stream_metric_change();
+-- INSERT should fire AFTER to gurantee to emit the final row value.
+DROP TRIGGER IF EXISTS stream_metric_raw_steps_trigger_i ON raw_steps;
+CREATE TRIGGER stream_metric_raw_steps_trigger_i
+    AFTER INSERT 
+                ON raw_steps
+                FOR EACH ROW EXECUTE PROCEDURE stream_metric_change();
 
-DROP TRIGGER IF EXISTS stream_metric_generic_metrics_trigger_d ON generic_metrics;
-CREATE TRIGGER stream_metric_generic_metrics_trigger_d
-    BEFORE DELETE ON generic_metrics
-    FOR EACH ROW EXECUTE PROCEDURE stream_metric_change();
+DROP TRIGGER IF EXISTS stream_metric_raw_validations_trigger_i ON raw_validations;
+CREATE TRIGGER stream_metric_raw_validations_trigger_i
+    AFTER INSERT
+                ON raw_validations
+                FOR EACH ROW EXECUTE PROCEDURE stream_metric_change();
 
--- Trigger for detecting metric permission scope changes derived from projects.workspace_id.
-CREATE OR REPLACE FUNCTION stream_metric_workspace_change_notify() RETURNS TRIGGER AS $$
-DECLARE
-    metric RECORD;
-    jmetric jsonb;
-BEGIN
-FOR metric IN
-SELECT
-    m.*,
-    p.workspace_id,
-    t.experiment_id
-FROM
-    projects p
-        INNER JOIN
-    experiment e ON e.project_id = p.id
-        INNER JOIN
-    trials t ON t.experiment_id = e.id
-        INNER JOIN
-    metrics m ON m.trial_id = t.id
-WHERE
-    p.id = NEW.id
-    LOOP
-        metric.seq = nextval('stream_metric_seq');
-        UPDATE metrics SET seq = metric.seq where id = metric.id;
-        jmetric = to_jsonb(metric);
-        PERFORM stream_metric_notify(jmetric, OLD.workspace_id, OLD.experiment_id, jmetric, NEW.workspace_id, NEW.experiment_id);
```

Comment:
yaaas delete it 
![soothing face small](https://github.com/determined-ai/determined/assets/31023784/dd430312-8957-4783-bcf3-f1d8e93e3bc6)